packages/bot-web-ui/src/components/flyout.jsx

```diff
@@ -111,30 +112,32 @@ const FlyoutContent = props => {
-                                            'dc-btn',
-                                            'dc-btn-effect',
-                                            'dc-btn--primary',
-                                            'flyout__button-new'
-                                        )}
-                                        onClick={button => {
-                                            const flyout_button = button;
-
-                                            // Workaround for not having a flyout workspace.
-                                            // eslint-disable-next-line no-underscore-dangle
-                                            flyout_button.targetWorkspace_ = Blockly.derivWorkspace;
-                                            flyout_button.getTargetWorkspace = () => {
-                                                // eslint-disable-next-line no-underscore-dangle
-                                                return flyout_button.targetWorkspace_;
-                                            };
-
-                                            callback(flyout_button);
-                                        }}
-                                    >
-                                        {node.getAttribute('text')}
-                                    </button>
+                                    <div key={index} className='flyout__new-variable-container'>
+                                        <Input
+                                            id='variable_name'
+                                            className='flyout__input'
+                                            data-lpignore='true'
+                                            type='text'
+                                            name='variable'
+                                            placeholder={localize('New variable name')}
+                                            ref={variable_name}
+                                        />
+                                        <button
+                                            key={`${cb_key}${index}`}
+                                            className={classNames(
+                                                'dc-btn',
+                                                'dc-btn-effect',
+                                                'dc-btn--primary',
+                                                'flyout__button-new'
+                                            )}
+                                            onClick={() => {
+                                                callback(variable_name.current.value);
+                                                variable_name.current.value = '';
+                                            }}
+                                        >
+                                            {node.getAttribute('text')}
+                                        </button>
+                                    </div>
```

Comment:
The original code would accommodate every button (in the toolbox's XML), this one however will turn any `<button>` into a create variable button, you should add a condition to handle these different cases (the `addCreateButton` programmatically adds a `<button>` to the toolbox).

- See https://developers.google.com/blockly/guides/configure/web/toolbox#buttons_and_labels

To check whether this is happening, you can add a snippet like below to the `toolbox.xml`.

```XML
<button text="Another button" callbackKey="someCallbackName"></button>
```